python-sdk/nuscenes/scripts/export_kitti.py

```diff
@@ -0,0 +1,347 @@
+            sd_record_lid = self.nusc.get('sample_data', lidar_token)
+            cs_record_cam = self.nusc.get('calibrated_sensor', sd_record_cam['calibrated_sensor_token'])
+            cs_record_lid = self.nusc.get('calibrated_sensor', sd_record_lid['calibrated_sensor_token'])
+
+            # Combine transformations and convert to KITTI format.
+            # Note: cam uses same conventions in KITTI and nuScenes.
+            lid_to_ego = transform_matrix(cs_record_lid['translation'], Quaternion(cs_record_lid['rotation']),
+                                          inverse=False)
+            ego_to_cam = transform_matrix(cs_record_cam['translation'], Quaternion(cs_record_cam['rotation']),
+                                          inverse=True)
+            velo_to_cam = np.dot(ego_to_cam, lid_to_ego)
+
+            # Convert from KITTI to nuScenes LIDAR coordinates, where we apply velo_to_cam.
+            velo_to_cam_kitti = np.dot(velo_to_cam, kitti_to_nu_lidar.transformation_matrix)
+
+            # Currently not used.
+            imu_to_velo_kitti = np.zeros((3, 4))  # Dummy values.
+            r0_rect = Quaternion(axis=[1, 0, 0], angle=0)  # Dummy values.
+
+            # Projection matrix.
+            p_left_kitti = np.zeros((3, 4))
+            p_left_kitti[:3, :3] = cs_record_cam['camera_intrinsic']  # Cameras are always rectified.
+
+            # Create KITTI style transforms.
+            velo_to_cam_rot = velo_to_cam_kitti[:3, :3]
+            velo_to_cam_trans = velo_to_cam_kitti[:3, 3]
+
+            # Check that the rotation has the same format as in KITTI.
+            assert (velo_to_cam_rot.round(0) == np.array([[0, -1, 0], [0, 0, -1], [1, 0, 0]])).all()
+            assert (velo_to_cam_trans[1:3] < 0).all()
+
+            # Retrieve the token from the lidar.
+            # Note that this may be confusing as the filename of the camera will include the timestamp of the lidar,
+            # not the camera.
+            filename_cam_full = sd_record_cam['filename']
+            filename_lid_full = sd_record_lid['filename']
+            # token = '%06d' % token_idx # Alternative to use KITTI names.
+            token_idx += 1
+
+            # Convert image (jpg to png).
+            src_im_path = os.path.join(self.nusc.dataroot, filename_cam_full)
+            dst_im_path = os.path.join(image_folder, sample_token + '.png')
+            if not os.path.exists(dst_im_path):
+                im = Image.open(src_im_path)
+                im.save(dst_im_path, "PNG")
+
+            # Convert lidar.
+            # Note that we are only using a single sweep, instead of the commonly used n sweeps.
+            src_lid_path = os.path.join(self.nusc.dataroot, filename_lid_full)
+            dst_lid_path = os.path.join(lidar_folder, sample_token + '.bin')
+            assert not dst_lid_path.endswith('.pcd.bin')
+            scan = np.fromfile(src_lid_path, dtype=np.float32)
+            points = scan.reshape((-1, 5))[:, :4].T
```

Comment:
A comment would be a good addition to this operation.

Reply:
Replaced with `LidarPointCloud.from_file(..)`